### PR TITLE
CBG-852 Stub out new sg-replicate API

### PR DESCRIPTION
**Pull request by adamcfraser**
Adds _replication and _replicationStatus endpoints.  Replication endpoints are wired up to cfg, replication status just returns a (formatted) static response.  Minimal request validation included, remainder is being left for CBG-766 and CBG-768.

### CODE REVIEW

File: rest/admin_api.go [hunk 1]
```diff
@@ -818,3 +818,92 @@ func (h *handler) handlePurge() error {
...
+	h.writeRawJSON(bytes)
+	return err
+}
+
+func (h *handler) putReplication() error {
+	replicationID := mux.Vars(h.rq)["replicationID"]
+	return h.upsertReplication(replicationID)
+}
+
+func (h *handler) upsertReplication(replicationID string) error {
```
Comment (bbrks) [inline rest/admin_api.go]: Just for my own benefit, is this split out from putReplication to be used for the future backwards compatibility on the `_replicate` endpoint?

File: rest/admin_api.go [hunk 2]
```diff
@@ -818,3 +818,92 @@ func (h *handler) handlePurge() error {
...
+
+	action := h.getQuery("action")
+	if action == "" {
+		return base.HTTPErrorf(http.StatusBadRequest, "Query parameter 'action' must be specified")
+	}
+
+	return h.db.SGReplicateMgr.PutReplicationStatus(replicationID, action)
+}
+
+// Delete tests need to cover 'not found'
```
Comment (bbrks) [inline rest/admin_api.go]: Just so it gets highlighted in the editor clearly

```suggestion
// TODO: Delete tests need to cover 'not found'
```

**Review by bbrks — CHANGES_REQUESTED** [review]
Just a few small comments

File: rest/admin_api.go (hunk 1 above)
Comment (adamcfraser) [reply rest/admin_api.go]: Ah, thanks for catching.  That is legacy from when PUT and POST were invoking different handler methods.  Now that they are all going through putReplication, I should avoid the extra call.

File: db/database.go [hunk 3]
```diff
@@ -414,26 +422,16 @@ func NewDatabaseContext(dbName string, bucket base.Bucket, autoImport bool, opti
 	return dbContext, nil
 }
 
+// InitCfgSG initializes Sync Gateway cluster config.  Used to manage shared import and sg-replicate config
+// between Sync Gateway nodes.  Required even if import is disabled and sgreplicate_enabled=false
+// to support REST API _replication/_replicationStatus requests.
 func (context *DatabaseContext) InitCfgSG() (err error) {
```
[the remark this answers is not in the document]
Comment (adamcfraser) [reply db/database.go]: Agreed - will remove.

File: rest/admin_api.go [hunk 4]
```diff
@@ -818,3 +818,92 @@ func (h *handler) handlePurge() error {
...
+
+// sg-replicate endpoints
+func (h *handler) getReplications() error {
+	replications, err := h.db.SGReplicateMgr.GetReplications()
+	if err != nil {
+		return err
+	}
+	bytes, err := base.JSONMarshal(replications)
+	h.writeRawJSON(bytes)
+	return err
```
[the remark this answers is not in the document]
Comment (adamcfraser) [reply rest/admin_api.go]: I agree.  In fact, makes me wonder whether the existing usages of writeRawJSON should be reviewed/changed.

File: rest/admin_api.go (hunk 2 above)
Comment (adamcfraser) [reply rest/admin_api.go]: I should probably just write the DELETE tests now - I'll do that.

**Review by bbrks — APPROVED** [review]
LGTM! :+1:

**Pull request merged**